transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/TransactionOutbox.java

```diff
@@ -284,6 +287,16 @@ public TransactionOutboxBuilder initializeImmediately(boolean initializeImmediat
       return this;
     }
 
+    /**
+     * @param submitImmediately If true, a task will be submitted immediately after scheduling.
+     * Defaults to true.
+     * @return Builder.
+     */
+    public TransactionOutboxBuilder submitImmediately(boolean submitImmediately) {
```

Comment:
At the moment, you can do `groupId(...)` without `submitImmediately(...)` and it looks like the behaviour is unexpected; because the task is submitted immediately, it ignores ordering.

Can we set `submitImmediately` to `true` automatically and refuse to allow `submitImmediately(false)` in combination with a `groupId`?

Better still: is this option required at all? I can't see that it adds anything other than a delay, if you're not using a `groupId`.  So why not just remove it?

Reply:
right, added it by analogy to other settings initially but now I see that it's mostly useless

Reply:
The submit immediately option might not be required to implement ordered processing. However, it could be useful if you want to separate transaction-processing compute from task-processing compute, i.e. only process tasks using `flush` from a separate JVM. This also makes the transaction-outbox library more suitable for use with serverless compute like AWS Lambda functions, where tasks cannot be processed in background threads.